staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go

```diff
@@ -286,6 +286,10 @@ func (s *Serializer) Decode(originalData []byte, gvk *schema.GroupVersionKind, i
 
 // Encode serializes the provided object to the given writer.
 func (s *Serializer) Encode(obj runtime.Object, w io.Writer) error {
+	if customEncoder, ok := obj.(runtime.CustomEncoder); ok {
+		return customEncoder.InterceptEncode(runtime.WithVersionEncoder{Encoder: s}, w)
```

Comment:
how is the InterceptEncode method expected to make use of the injected encoder? if I'm following this correctly, it can't call `encoder.Encoder(self)` or we'll recurse. Does that mean every implementation of InterceptEncode must be implemented as a wrapper object around a normal encodable object?

Reply:
That was my intention - yes. Do you have a better suggestion to do that?